x-pack/plugins/streams/server/routes/streams/delete.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+import { IScopedClusterClient } from '@kbn/core-elasticsearch-server';
+import { Logger } from '@kbn/logging';
+import {
+  DefinitionNotFound,
+  ForkConditionMissing,
+  IndexTemplateNotFound,
+  SecurityException,
+} from '../../lib/streams/errors';
+import { createServerRoute } from '../create_server_route';
+import { syncStream, readStream, deleteStreamObjects } from '../../lib/streams/stream_crud';
+import { MalformedStreamId } from '../../lib/streams/errors/malformed_stream_id';
+import { getParentId } from '../../lib/streams/helpers/hierarchy';
+
+export const deleteStreamRoute = createServerRoute({
+  endpoint: 'DELETE /api/streams/{id} 2023-10-31',
+  options: {
+    access: 'public',
+    security: {
```

Comment:
Availability is not documented yet, but in this case we can set this (and other  APIs) to `experimental`

```suggestion
  options: {
    access: 'public',
     availability: {
       stability: 'experimental'
     },
    security: {
```

Reply:
Thanks, set to experimental